.github/scripts/ci_tests.sh

```diff
@@ -11,20 +11,27 @@ TEST_RUN_E2E_TESTS=${1:-1}
 TEST_CORES=${2:-2}
 TEST_JOB_COUNT=${3:-2}
 
-echo -e "Building tests"
-cmake tests -Bbuild
+echo -e "::group::Building unit tests"
+cmake -Dbuild_ut_tests=ON -Bbuild .
 cmake --build build -- --jobs=$TEST_CORES
-cd build
+pushd build
 
-echo -e "Running unit tests"
+echo -e "::group::Running unit tests"
 ctest -j $TEST_JOB_COUNT -C "debug" --output-on-failure --schedule-random -T test
 
+echo -e "::group::Code coverage"
+gcovr -r $(pwd) -f ../source/.*.c
+
+popd
+
+echo -e "::group::Building e2e tests"
```

Comment:
E2E test running it two step process .
1. Building device side binary: require some config setting up and then building with those config.
2. Compile service side scripts and run them.

These both steps are coded in the script (./tests/e2e/run.sh). You do not need to compile code here. Check this file
https://github.com/Azure/azure-iot-middleware-freertos/blob/cbb4585fc6a903ea90264556cd23bcede882e74f/tests/e2e/run.sh#L40

Possibly this special echo can be done in that file.

Reply:
ahh I see didn't realize the script was building too. I thought it was just running the executable.